### PR TITLE
Improve accessibility by associating `for` with `id` instead of `name`

### LINKED ISSUE #214: for/id associations for ThreePartDate
From VT's accessibility audit, relating to the ThreePartDateInput.

> On step 11, the “Month”, “Day”, and “Year” < label > elements are disassociated from their inputs
(seemingly deliberately, judging by the word “ignore” in their for attributes). As a result, screen reader
users cannot hear these labels. We recommend creating proper for / id associations. 3-Moderate

Here's an example chunk of the ThreePartDate HTML for reference.

```html
<div class="col-md-8 dafieldpart">
    <div class="al_three_parts_date form-row row">
        <div class="col col-month">
            <label for="_ignore_ZGF0ZV9vZl9ncmFkdWF0aW9u_month">Month</label>
            <select class="form-select al_field month ZGF0ZV9vZl9ncmFkdWF0aW9u" id="ZGF0ZV9vZl9ncmFkdWF0aW9u_month" name="_ignore_ZGF0ZV9vZl9ncmFkdWF0aW9u_month" aria-describedby="ZGF0ZV9vZl9ncmFkdWF0aW9u-error" aria-invalid="false">
                <option value=""></option>
                <option value="01">January</option>
                <option value="02">February</option>
                ...
            </select>
        </div>
        <div class="col col-3 col-day">
            <label for="_ignore_ZGF0ZV9vZl9ncmFkdWF0aW9u_day">Day</label>
            <input class="form-control al_field day ZGF0ZV9vZl9ncmFkdWF0aW9u" type="number" min="1" inputmode="numeric" id="ZGF0ZV9vZl9ncmFkdWF0aW9u_day" name="_ignore_ZGF0ZV9vZl9ncmFkdWF0aW9u_day" aria-describedby="ZGF0ZV9vZl9ncmFkdWF0aW9u-error">
        </div>
        <div class="col col-3 col-year">
            <label for="_ignore_ZGF0ZV9vZl9ncmFkdWF0aW9u_year">Year</label>
            <input class="form-control al_field year ZGF0ZV9vZl9ncmFkdWF0aW9u" type="number" min="1" inputmode="numeric" id="ZGF0ZV9vZl9ncmFkdWF0aW9u_year" name="_ignore_ZGF0ZV9vZl9ncmFkdWF0aW9u_year" aria-describedby="ZGF0ZV9vZl9ncmFkdWF0aW9u-error">
        </div>
        <div id="al_ZGF0ZV9vZl9ncmFkdWF0aW9u_error" class="al_error"></div>
    </div>
    <input value="02/3/2040" alt="Input box" class="form-control daThreePartsDate" data-almin="11/28/2023" data-almax="12/31/2040" type="hidden" name="ZGF0ZV9vZl9ncmFkdWF0aW9u" id="ZGF0ZV9vZl9ncmFkdWF0aW9u" style="display: none;" aria-hidden="true">
</div>
```

I'll have to look into what DA and HTML normally do with `name` compared to `id`, but I might need your help / perspective when building this @plocket

### PR DESCRIPTION
**Pull request by plocket**
Closes #214. No rush on my part. I've tested it a bit myself to make sure basic behavior hasn't changed, though I don't think I have the tools I need to test the accessibility properly.

Not sure if we want to increment those package versions.

### CODE REVIEW

**Review by nonprofittechy — APPROVED** [review]
Just need to fix formatting; I'll merge that PR

**Comment by plocket** [comment]
I think Bryce wanted to test some things before merging this

**Comment by nonprofittechy** [comment]
> I think Bryce wanted to test some things before merging this

I meant merge the PR from actions/Black, but that doesn't exist. Looks like we need to point to Bryce's new action in ALActions for formatting; the one this repo points to is broken.

File: docassemble/ALToolbox/ThreePartsDate.py [hunk 1]
```diff
@@ -175,11 +175,11 @@
   */
   var $col = $('<div class="col col-month">');
   
-  let id = date_id + '_month';
-  // '_ignore' prevents the field from being submitted, avoiding a da error
-  let name =  '_ignore_' + id;
+  var id = '_ignore_' + date_id + '_month';
```
Comment (BryceStevenWilley) [inline docassemble/ALToolbox/ThreePartsDate.py:178]: Any reason you changed `let` to `var` here, but not elsewhere? IMO it's either we do need it for compatibility, or we don't.

**Review by BryceStevenWilley — CHANGES_REQUESTED** [review]
Got to testing it out, and it does seem to associate the "Month" / "Day" / "Year" labels better with their inputs, so success! However, I think the conversation we started having in #164 comes into play too, as there are some issues with the original field label.

In Chromium, I get this error:

```
The label's for attribute doesn't match any element id. This might prevent the
browser from correctly autofilling the form and accessibility tools from working correctly.

To fix this issue, make sure the label's for attribute references the correct id of a form field.
```

which happens because we set that input to be hidden, so most accessibility tools won't include it as a part of the page's content, and will assume that the label points to nothing.

I think that we could make the whole widget a [fieldset](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset), with that label becoming the `legend`. I manually edited that HTML in the browser tools, and it's more clearly associated with the inputs, which is great, essentially just changing the `div.da-field-container-datatype-ThreePartsDate` to a `fieldset`, and changing the `label` tag to a `legend`. My HTML looks like this:

```html
<fieldset class="da-container da-form-group row da-field-container da-field-container-datatype-ThreePartsDate">
    <legend for="ZGF0ZV9vZl9ncmFkdWF0aW9u" class="col-md-4 col-form-label da-form-label datext-right">
        When will your high school graduation be?
    </legend>
    <div class="col-md-8 dafieldpart">
        (unchanged)
        ...
</fieldset>
```

I'm not sure how much harder these changes would be; if you think they're a lot harder, we can merge this and take another shot at them later (and to be extra clear, these issues are also present in main, so there's nothing wrong with this PR). But if you think that it'd be easy to add while this is fresh in our minds, I think it'd be a significant improvement.

File: docassemble/ALToolbox/ThreePartsDate.py (hunk 1 above)
Comment (plocket) [reply docassemble/ALToolbox/ThreePartsDate.py:178]: You're right, we should be able to use `let` now. It's just an old habit. I'll change it back 👍

**Comment by plocket** [comment]
> I'm not sure how much harder these [fieldset] changes would be; if you think they're a lot harder, we can merge this and take another shot at them later

I do believe the `fieldset` conversation is a more complex one and [added to the discussion in #164](https://github.com/SuffolkLITLab/docassemble-ALToolbox/issues/164#issuecomment-1856049920). We should probably save it for a separate discussion and PR. Maybe a deep dive? I've added it to the list in the doc for whenever the next deep dive happens.

File: docassemble/ALToolbox/ThreePartsDate.py [hunk 2]
```diff
@@ -175,11 +175,11 @@
...
   
-  let id = date_id + '_month';
-  // '_ignore' prevents the field from being submitted, avoiding a da error
-  let name =  '_ignore_' + id;
+  let id = '_ignore_' + date_id + '_month';
+  // '_ignore' in the name prevents the field from being submitted, avoiding a da error
+  let name =  id;
   
-  let $label = $('<label for="' + name + '">{month}</label>');
+  var $label = $('<label for="' + id + '">{month}</label>');
```
Comment (BryceStevenWilley) [inline docassemble/ALToolbox/ThreePartsDate.py:182]:
```suggestion
  let $label = $('<label for="' + id + '">{month}</label>');
```

**Pull request merged**